dashboard/dashboard.py

```diff
@@ -1,12 +1,53 @@
+# pylint: disable=C0103
 """
 This is the main dashboard for the Games Tracker Streamlit application.
 It includes data visualisations on the information collected from Steam, GOG and Epic,
 which is stored in an AWS RDS database.
 """
+from datetime import datetime
 
 import streamlit as st
 
 # Page configuration
 st.set_page_config(layout="wide")
 
-st.title("Games Tracker")
+cols = st.columns([2, 1, 1, 0.5])
+with cols[0]:
+    st.title("Games Tracker")
+
+# Create a dropdown selection
+with cols[1]:
+    platform_selection = st.selectbox(
+        "Select a platform:",
+        options=["-All-", "Windows", "Mac", "Linux"])
+
+with cols[2]:
+    # Get today's date to set as the maximum selectable date
+    today = datetime.now().date()
+
+    # Set a specific minimum date (e.g., October 7, 2024)
+    min_date = datetime(2024, 10, 7).date()
+
+    # Create a date range selector with "All time" as the default range
+    # Default to the range from min_date to today
+    default_start_date = min_date
+    default_end_date = today
+
+    # Create the date input and store the output
+    date_range = st.date_input(
```

Comment:
This is some fancy stuff right here. Using date ranges will make the Pandas stuff easier too.